sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/CommonUtil.kt

```diff
@@ -0,0 +1,246 @@
+     */
+    fun isEqual(first: String, second: String): Boolean {
+        return first == second
+    }
+
+    /**
+     * Check the string whether null or empty
+     *
+     * @param rawString the raw string
+     * @return true if null or empty otherwise false
+     */
+    fun isNullOrEmpty(rawString: String?): Boolean {
+        return rawString.isNullOrEmpty()
+    }
+
+    /**
+     * Get the encoded Sha256 digest string
+     *
+     * @param rawString the raw string.
+     * @return the encoded string
+     */
+    fun getSha256HashData(rawString: String?): String? {
+        return if (rawString == null) {
+            null
+        } else try {
+            val md = MessageDigest.getInstance("SHA-256")
+            val byteData = md.digest(getEncodedByteArray(rawString, "utf-8"))
+            val sb = StringBuilder()
+            for (b in byteData) {
+                val hex = Integer.toHexString(0xFF and b.toInt())
+                if (hex.length == 1) {
+                    sb.append('0')
+                }
+                sb.append(hex)
+            }
+            sb.toString()
+        } catch (e: NoSuchAlgorithmException) {
+            null
+        }
+    }
+
+    /**
+     * Check the device to make sure it has the Google Play Services APK.
+     *
+     *
+     * If it doesn't, display a dialog that allows users to download the APK from the Google Play
+     * Store or enable it in the device's system settings.
+     *
+     * @param context non null activity, used to display update dialog if necessary.
+     * @return true if play services are available, false otherwise
+     */
+    fun checkPlayServices(context: Activity): Boolean {
+        val googleApi = GoogleApiAvailability.getInstance()
+        val result = googleApi.isGooglePlayServicesAvailable(context)
+        if (result != ConnectionResult.SUCCESS) {
+            if (googleApi.isUserResolvableError(result)) {
+                googleApi.getErrorDialog(context, result, PLAY_SERVICES_RESOLUTION_REQUEST)?.show()
+            }
+            return false
+        }
+        return true
+    }
+
+    /**
+     * Get the UTC date format from the simple date format, i.e. "yyyy-MM-dd'T'HH:mm:ssX"
+     *
+     * @return the date format
+     */
+    @SuppressLint("SimpleDateFormat")
+    fun getUTCDateFormat(): DateFormat {
+        val dateFormat: DateFormat = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssX")
+        } else {
+            SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss")
+        }
+        val timeZone = TimeZone.getTimeZone("UTC")
+        dateFormat.timeZone = timeZone
+        return dateFormat
+    }
+
+    /**
+     * Get the UTC date from a sample date string
+     *
+     * @param dateString the date string.
+     * @return the date object based on the date string
+     */
+    @SuppressLint("SimpleDateFormat")
+    fun getUTCDate(dateString: String?): Date? {
```

Comment:
Can we make this as extension of `Date` class?

Reply:
I tried but not working, please give some idea/example

Reply:
Better to move this to `String` extension